### PR TITLE
Migrate S2251 to custom eslint-based implementation (for-loop-increment-sign)

### LINKED ISSUE #2100: Migrate S2251 to eslint frontend (new rule for TS)

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
Fixes  #2100

### CODE REVIEW

File: eslint-bridge/tests/rules/for-loop-increment-sign.test.ts [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    },
+  ],
+  invalid: [
+    {
+      code: `
+            for (i = 0; i < 5; i--) {}
+            `,
+      errors: [
+        {
+          message: `{"message":"\\"i\\" is decremented and will never reach its stop condition.","secondaryLocations":[{"column":24,"line":2,"endColumn":29,"endLine":2}]}`,
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/tests/rules/for-loop-increment-sign.test.ts:165]: I think you can use `^^^^` LITS convention in the code snippet for primary and secondary locations, just for the sake of clarity

**Review by andrea-guarino-sonarsource — APPROVED** [review]
LGTM, some minor comments

File: eslint-bridge/tests/rules/for-loop-increment-sign.test.ts (hunk 1 above)
Comment (guillaume-dequenne-sonarsource) [reply eslint-bridge/tests/rules/for-loop-increment-sign.test.ts:165]: Just so I'm sure I understood well: that would improve readability but would not actually be asserted, right?

Comment (andrea-guarino-sonarsource) [reply eslint-bridge/tests/rules/for-loop-increment-sign.test.ts:165]: yes exactly, only for readability

File: eslint-bridge/src/rules/for-loop-increment-sign.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      if (incrementDirection !== null && isSameIdentifier(rhs.left, lhs)) {
+        incrementDirection = rhs.operator === '-' ? -incrementDirection : incrementDirection;
+        return ForLoopIncrement.increment(assignmentExpression, lhs, incrementDirection);
+      }
+    }
+    return null;
+  }
+}
+
+function directionFromValue(expression: estree.Expression): number {
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply eslint-bridge/src/rules/for-loop-increment-sign.ts:141]: I actually get a warning from VSCode if I don't, and I must admit I feel it's nicer to have it (don't we love type hints? 😄)

**Pull request merged**